twill-core/src/main/java/org/apache/twill/kafka/client/KafkaOffsetProvider.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.twill.kafka.client;
+
+/**
+ * Define interface that could provide a method to check whether a message meets a given condition. If the condition is
+ * not met, the method will return the next offset to continue searching for the message meeting this condition.
+ */
+public interface KafkaOffsetProvider {
+
+  /**
+   * Check whether a message meets a given condition. If the condition is not met, return the next offset to
```

Comment:
It will be provided in the constructor. For instance, the timestamp will be passed as a parameter in the constructor. Other parameters can also be passed instead, and they can be used in `getCandidateOffset` method